core/src/main/scala/cr/pulsar/internal/TypedMessageBuilderOps.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Chatroulette
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cr.pulsar.internal
+
+import cr.pulsar.Producer.{ MessageKey, ShardKey }
+import org.apache.pulsar.client.api.TypedMessageBuilder
+
+object TypedMessageBuilderOps {
+  implicit class TMBOps(value: TypedMessageBuilder[Array[Byte]]) {
```

Comment:
Let's make it a value class? e.g. `extends AnyVal`.

https://docs.scala-lang.org/overviews/core/value-classes.html#extension-methods

Reply:
And I guess this is only used internally, right? If so, we should make it `private[internal]` or so.